firebase-firestore/src/test/java/com/google/firebase/firestore/spec/SpecTestCase.java

```diff
@@ -654,15 +674,22 @@ private void doWatchEntity(JSONObject watchEntity) throws Exception {
     }
   }
 
-  private void doWatchFilter(JSONArray watchFilter) throws Exception {
-    List<Integer> targets = parseIntList(watchFilter.getJSONArray(0));
+  private void doWatchFilter(JSONObject watchFilter) throws Exception {
+    List<Integer> targets =
```

Comment:
Could you explain why you had to add this check for `watchFilter.has("targetIds")`? Do you think it warrants an inline comment to explain?

Reply:
While building spec tests, targetIds and keys could be empty array, added the check for precaution. but can be removed. I will remove both.